internal/secretref/secret_ref.go

```diff
@@ -28,6 +30,29 @@ func ReferencesNonExistentSecret(ctx context.Context, client client.Reader, gett
 	return false
 }
 
+func SanitizeTlSValueOrSecret(secretData map[string][]byte, tlsKeyVariable, tlsCertVariable string) map[string][]byte {
```

Comment:
I think we can improve multiple things about this function:
1. Map arguments are references in Go, so no need to return the same map
2. Abou the naming
  a. I cannot distinguish by the name what is the difference between this and the `validateAndSanitizeTLSSecret` helper.
  b. I do not understand the `Or` in the name. This function handles only one case
  c. Replace TlS by TLS in the name
3. I still think this function should not be part of the secretref package. It works with a map, but does not have anything to do with a Secret(-reference)

Reply:
I removed this function and simplified the code